client/src/components/Event.tsx

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+
+// Definice typů pro uživatele a datumy
+type UserRecord = {
+  name: string;
+  answer: "yes" | "no" | "if-needed";
+};
+
+type DateRecord = {
+  timestamp: number;
+  records: UserRecord[];
+};
+
+type EventProps = {
+  location?: string;
+  id: string;
+  title: string;
+  dates: DateRecord[];
+};
+
+// Mapování odpovědí na symboly
+const answerSymbols: Record<UserRecord["answer"], string> = {
+  yes: "✅",
+  no: "❌",
+  "if-needed": "🤔",
+};
+
+// Komponenta pro vykreslení řádku tabulky
+const VotingRow: React.FC<{ user: UserRecord; dates: DateRecord[] }> = ({ user, dates }) => (
+  <tr>
+    <td>{user.name}</td>
+    {dates.map((date) => {
+      const userVote = date.records.find((record) => record.name === user.name);
+      return <td key={date.timestamp}>{userVote ? answerSymbols[userVote.answer] : "—"}</td>;
+    })}
+  </tr>
+);
+
+// Komponenta pro vykreslení celé tabulky hlasování
+const VotingTable: React.FC<{ dates: DateRecord[] }> = ({ dates }) => {
```

Comment:
Tady bych asi volil trochu jinou strukturu:
- urcite komponentu Event dat do slozky src/components/Event (takze src/components/Event/Event.tsx , test to same)
- asi bych se priklanel (byt tady neni uplne nutne vzhledem k nizke komplexite) dat komponenty `VotingTable` a `VotingRow` taky do vlastnich slozek (src/components/VotingTable/VotingTable.tsx, src/components/VotingTable/VotingRow.tsx) pripadne je dat jen do vlastnich souboru a ty umistit to src/components/Event slozky

Ale neni to nic, co by jsi musela nutne menit, jen do budoucna dobre to vzit v potaz :)